odoo/tools/convert.py

```diff
@@ -719,12 +725,19 @@ def env(self):
     def noupdate(self):
         return self._noupdate[-1]
 
+    def next_sequence(self):
+        if self._sequences[-1] is None:
+            return None
+        self._sequences[-1] += 10
+        return self._sequences[-1]
+
     def __init__(self, cr, module, idref, mode, noupdate=False, xml_filename=None):
         self.mode = mode
         self.module = module
         self.envs = [odoo.api.Environment(cr, SUPERUSER_ID, {})]
         self.idref = {} if idref is None else idref
         self._noupdate = [noupdate]
+        self._sequences = []
```

Comment:
Those stacks should have the same size, otherwise this is way too bug prone.

Reply:
I'm not sure what `noupdate` is doing. If I understand the code, it will never be read since you will always go through at least one root node, thus appending a new value to the list, before parsing some record nodes.